### DISCUSSION TITLE
Retrieve current input file line in Lua filter

### DISCUSSION THREAD

**Discussion by unitscan** [jgm/pandoc#9903] [Q&A]
From https://pandoc.org/lua-filters:

```
function Header(el)
    if el.level == 1 then
      return el:walk {
        Str = function(el)
            return pandoc.Str(text.upper(el.text))
        end
      }
    end
end
```
It is possible to know the number(s) of the line(s) of the input file at which el.level == 1?

**Comment by jgm** [comment]
This is not generally stored in the AST.  Exception:  when the `sourcepos` extension is used
https://pandoc.org/MANUAL.html#extension-sourcepos
but it is only available for commonmark and variants (gfm).
